keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -32,218 +31,110 @@ status: provisional
-
-Components should plumb the context through reconciliation functions, rather than storing and looking up trace contexts globally so that each attempt to reconcile desired and actual state uses the context associated with _that_ desired state through the entire attempt.  If multiple components are involved in reconciling a single object, one may act on the new trace context before the other, but each trace is still representative of the work done to reconcile to the corresponding desired state. Given this model, we guarantee that each trace contains the actions taken to reconcile toward a single desired state.
-
-High-level processes, such as starting a pod or restarting a failed container, could be interrupted before completion by an update to the desired state. While this leaves a "partial" trace for the first process, it is the most accurate representation of the work and timing of reconciling desired and actual state.
-
-#### End-User Behavior
-
-Add a new `--trace` argument to `kubectl`, which generates a new trace context, sets the trace context to be sampled, attaches the context to all modified objects, and uses the context when sending requests to the API Server.  The option is disabled by default.  Note that by attaching a trace context to the initial object creation, this will cause all object modification done by controllers to propagate the context through to all changes made by the system that are driven by the initial user action.
-
-Add `context.Context` arguments to k8s.io/client-go client functions.  This will allow users and components to associate API calls with the context of the involved object.  In some cases, such as object creation, we can automatically attach the SpanContext of the provided context to the created object, making propagation simpler.
+### Tracing API Requests
 
-This also enables kubernetes to be a composable part of a larger system. For example, if an end-user's service creates a pod as part of handing a request, it could do:
-```golang
-ctx, span := trace.StartSpan(preexistingContext, “create-my-pod”)
-defer span.End()
-pod, err := c.CoreV1().Pod(myPod.Namespace).Create(ctx, myPod)
-if err != nil {
-    return err
-}
-waitForPodToBeRunning(ctx, myPod)
-return nil
-```
+We will wrap the API Server's http server and http clients with [othttp](https://github.com/open-telemetry/opentelemetry-go/tree/master/plugin/othttp) to get spans for incoming and outgoing http requests, and add the [otgrpc](https://github.com/grpc-ecosystem/grpc-opentracing/tree/master/go/otgrpc) DialOption to the etcd grpc client.  This generates spans for all sampled incoming requests and propagates context with all client requests.
```

Comment:
Where in the filter stack would you put that?

Reply:
It would be a non-mutating filter for incoming requests, so it shouldn't interfere with any of the existing filters.  It should definitely go after Authn/z to make sure unauthenticated clients can't generate lots of spans, and should come before the panic filter.  Anywhere in that range should be fine.  I would probably put it above or below WithRequestInfo, as that also attaches something to the context.Context.